be/src/http/action/compaction_action.cpp

```diff
@@ -138,7 +142,13 @@ Status CompactionAction::do_compaction(uint64_t tablet_id, const string& compact
             if (rowset_ids.empty()) {
                 return Status::InvalidArgument(fmt::format("empty argument. rowset_ids:{}", rowset_ids_string));
             }
-            RETURN_IF_ERROR(tablet->updates()->compaction(mem_tracker, rowset_ids));
+            res = tablet->updates()->compaction(mem_tracker, rowset_ids);
+        }
+        if (!res.ok()) {
+            StarRocksMetrics::instance()->update_compaction_request_failed.increment(1);
+            LOG(WARNING) << "failed to perform update compaction. res=" << res.to_string()
```

Comment:
`res.get_error_msg()` instead of `res.to_string()`

Reply:
Resolved.